orangecontrib/text/widgets/owtwitter.py

```diff
@@ -0,0 +1,422 @@
+    text_area_height = 80
+
+    MISSED_KEY = 1
+    MISSED_KEY_MESSAGE = 'Please provide a valid API key in order to get the data.'
+    MISSED_QUERY = 2
+    MISSED_QUERY_MESSAGE = 'Please input query words.'
+    API_ERROR = 3
+
+    tweets_info = 'Tweets on output: {}'
+
+    CONTENT, AUTHOR, CONTENT_AUTHOR = 0, 1, 2
+    recent_api_keys = Setting([])
+    word_list = Setting('')
+    mode = Setting(0)
+    limited_search = Setting(True)
+    max_tweets = Setting(100)
+    language = Setting(None)
+    advance = Setting(False)
+    include = Setting(False)
+    includeON = Setting(True)
+
+    error_signal = QtCore.pyqtSignal(str)
+    finish_signal = QtCore.pyqtSignal()
+    progress_signal = QtCore.pyqtSignal(int)
+    start_signal = QtCore.pyqtSignal()
+
+    attributes = [
+        ('Author', ('author_screen_name', )),
+        ('Content', ('text',)),
+        ('Date', ('created_at',)),
+        ('Language', ('lang',)),
+        ('In Reply Ro', 'in_reply_to_user_id'),
+        ('Number of Likes', ('favorite_count',)),
+        ('Number of Retweets', ('retweet_count',)),
+        ('Coordinates', ('coordinates_longitude', 'coordinates_latitude', 'place')),
+        ('Tweet ID', ('id',)),
+        ('Author ID', ('author_id',)),
+        ('Author Name', ('author_name', )),
+        ('Author Description', ('author_description',)),
+        ('Author Statistic', ('author_statuses_count', 'author_favourites_count',
+                              'author_friends_count', 'author_followers_count',
+                              'author_listed_count')),
+        ('Author Verified', ('author_verified',)),
+    ]
+    corpus_variables = Setting([val for _, val in attributes[:3]])
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.key = self.recent_api_keys[0] if self.recent_api_keys else None
+        self.api = None
+
+        self.controlArea.layout().setContentsMargins(0, 15, 0, 0)
+
+        # Set API key button.
+        key_dialog_button = gui.button(self.controlArea, self, 'Twitter API Key',
+                                       callback=self.open_key_dialog,
+                                       tooltip="Set the API key for this widget.")
+        key_dialog_button.setFocusPolicy(QtCore.Qt.NoFocus)
+
+        # Query
+        self.query_box = gui.hBox(self.controlArea, 'Query')
+
+        # Queries configuration
+        layout = QtGui.QGridLayout()
+        layout.setSpacing(7)
+
+        # Query
+        row = 0
+        query_edit = ListEdit(self, 'word_list', self)
+        query_edit.setFixedHeight(self.text_area_height)
+        layout.addWidget(QtGui.QLabel('Query word list:'), row, 0, 1, self.label_width)
+        layout.addWidget(query_edit, row, self.label_width, 1, self.widgets_width)
+
+        # Search mode
+        row += 1
+        mode = gui.comboBox(self, self, 'mode')
+        mode.addItem('Content')
+        mode.addItem('Author')
+        mode.addItem('Content & Author')
```

Comment:
Is it possible for one mode to support searching by [advanced query](https://twitter.com/search-advanced) like "[sanders -clinton lang:ru since:2015-06-01 :)](https://twitter.com/search?q=sanders%20-clinton%20lang%3Aru%20since%3A2015-06-01%20%3A%29&src=typd)"?